### DISCUSSION TITLE
why git clone needed for dvc pull?

### DISCUSSION THREAD

**Discussion by eangius** [treeverse/dvc#7639] [General]
Hello experts, 👋🏼 

Would anyone know why do `dvc pull` commands need to run a `git clone` under-the-hood? I stumbled across this in one of my environments which I found a bit surprising & this got me aching to better understand DVC's design to better integrate with git-submodules.

Thanks,

**Comment by pmrowla** [comment]
If your repo has imports, DVC needs to clone the the repo you are importing from

**Reply by eangius** [reply]
Awesome! This explains it perfectly.
